testing/src/test/java/io/stargate/it/cql/BoundStatementTest.java

```diff
@@ -0,0 +1,111 @@
+package io.stargate.it.cql;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.datastax.oss.driver.api.core.cql.BoundStatement;
+import com.datastax.oss.driver.api.core.cql.PreparedStatement;
+import com.datastax.oss.driver.api.core.cql.Row;
+import com.datastax.oss.driver.api.core.cql.SimpleStatement;
+import com.datastax.oss.driver.api.core.servererrors.InvalidQueryException;
+import io.stargate.it.storage.ClusterConnectionInfo;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Note: all tests related to unset values assume that the protocol version is always 4 or higher.
+ */
+public class BoundStatementTest extends JavaDriverTestBase {
+
+  private static final String KEY = "test";
+  private static final int VALUE = 7;
+
+  public BoundStatementTest(ClusterConnectionInfo backend) {
+    super(backend);
+  }
+
+  @BeforeEach
+  public void setupSchema() {
+    // table where every column forms the primary key.
+    session.execute("CREATE TABLE IF NOT EXISTS test (k text, v int, PRIMARY KEY(k, v))");
```

Comment:
If I can propose something upfront:
stargate + backend == cluster, so
I think we should wrap DDL with something that will:
- run DDL
- wait for schema agreement

This should keep tests more stable, at least that is what I learned with QA2 and dtests.

Reply:
The driver already waits for schema agreement before completing a DDL query, so there's no need to do it explicitly.

But overall I agree that the test infrastructure should be abstracted more. Eventually I'd like to move all `CqlSession` management to a JUnit extension, but that's not the top priority right now.

Reply:
TIL. Good thing.

Reply:
With one notice....

So, we have stargate node and backend node.
Driver is aware only of stargate node.
So schema agreement does not mean much in this case iiuc.